app/src/main/kotlin/org/fossify/notes/activities/MainActivity.kt

```diff
@@ -1557,4 +1564,24 @@ class MainActivity : SimpleActivity() {
             updateWidgets()
         }
     }
+
+    private fun checkReadOnlyState() {
+        getCurrentFragment()?.apply {
+            if (this is TextFragment) {
+                (this as TextFragment).getNotesView().isEnabled = !mCurrentNote.isReadOnly
+            }
+        }
```

Comment:
Since you are already checking the type, there's no need to cast it explicitly:

```suggestion
        getCurrentFragment()?.apply {
            if (this is TextFragment) {
                getNotesView().isEnabled = !mCurrentNote.isReadOnly
            }
        }
```

or using the safe cast operator:

```suggestion
        (getCurrentFragment() as? TextFragment)
            ?.getNotesView()?.isEnabled = !mCurrentNote.isReadOnly
```